game-core/src/main/java/games/strategy/triplea/delegate/EndTurnDelegate.java

```diff
@@ -216,12 +217,28 @@ private String addUnitCreatedResources(final IDelegateBridge bridge) {
   }
 
   /**
-   * Find the PU value of national objectives that are currently met.
+   * Find the resources generated by national objectives and triggers that are currently met.
    */
-  public static int findNationalObjectivePus(final PlayerID player, final GameData data, final IDelegateBridge bridge) {
+  public static IntegerMap<Resource> findNationalObjectiveAndTriggerResources(final PlayerID player,
+      final GameData data) {
+    final IDelegateBridge bridge = new ObjectiveDummyDelegateBridge(data);
+
+    // Find and test all the conditions for triggers and national objectives
+    final Set<TriggerAttachment> triggers = new HashSet<>();
     final List<RulesAttachment> objectives = new ArrayList<>();
     final HashMap<ICondition, Boolean> testedConditions =
-        testNationalObjectivesAndTriggers(player, data, bridge, new HashSet<>(), objectives);
+        testNationalObjectivesAndTriggers(player, data, bridge, triggers, objectives);
+
+    // Find triggers value
+    IntegerMap<Resource> resources = new IntegerMap<>();
```

Comment:
`resources` could be made `final` with an `else` block below that initializes it to an empty map.

Reply:
Updated.